library/cwm/src/lib/cwm/multi_status_selector.rb

```diff
@@ -0,0 +1,470 @@
+        # TRANSLATORS: help text for an automatically selected check box
+        # (it has a different look that a user selected check box)
+        help_text << "#{icon_for(AUTO_SELECTED)} = #{_("Auto selected")}"
+        help_text << "</p>"
+        help_text
+      end
+
+      # Returns the icon to be used for an item with given status and state
+      #
+      # @see .value_for
+      #
+      # @param status [Symbol] the item status (e.g., :selected, :registered, :auto_selected)
+      # @param mode [String] the running mode, "normal" or "inst"
+      # @param state [String] the item state, "enabled" or "disabled"
+      #
+      # @return [String] an <img> tag when running in GUI mode; plain text otherwise
+      def self.icon_for(status, mode: "normal", state: "enabled")
+        value = value_for(status)
+
+        if Yast::UI.TextMode
+          value
+        else
+          # an image key looks like "inst:[a]:enabled"
+          image_key = [mode, value, state].join(":")
+
+          "<img src=\"#{IMAGES_DIR}/#{IMAGES[image_key]}\">"
+        end
+      end
+
+      # Returns the status string representation
+      #
+      # @param status [Symbol] the status identifier
+      #
+      # @return [String] the status text representation
+      def self.value_for(status)
+        case status
+        when SELECTED
+          "[x]"
+        when AUTO_SELECTED
+          "[a]"
+        else
+          "[ ]"
+        end
+      end
+
+      # Toggles the current status
+      def toggle
+        @status = selected? ? UNSELECTED : SELECTED
+      end
+
+      # Determines if the item is enabled or not
+      #
+      # @return [Boolean] true when item is enabled; false otherwise
+      def enabled?
+        true
+      end
+
+      # Whether item is selected
+      #
+      # @return [Boolean] true if the status is selected; false otherwise
+      def selected?
+        status == SELECTED
+      end
+
+      # Whether item is not selected
+      #
+      # @return [Boolean] true if the status is not selected; false otherwise
+      def unselected?
+        [SELECTED, AUTO_SELECTED].none?(status)
+      end
+
+      # Whether item is auto selected
+      #
+      # @return [Boolean] true if the status is auto selected; false otherwise
+      def auto_selected?
+        status == AUTO_SELECTED
+      end
+
+      # Sets the item as selected
+      def select!
+        @status = SELECTED
+      end
+
+      # Sets the item as not selected
+      def unselect!
+        @status = UNSELECTED
+      end
+
+      # Sets the item as auto-selected
+      def auto_select!
+        @status = AUTO_SELECTED
+      end
+
+      # Returns richtext representation for the item
+      #
+      # Basically, an string containing two <a> or <span> tags, depending on the #enabled? method.
+      # One for the check box input and another for the label.
+      #
+      # @return [String] the item richtext representation
+      def to_s
```

Comment:
I am not sure if to_s is ideal here as it can be confused. What about to_richtext or to_html ?

Reply:
Great idea. I'll use `to_richtext` because as far as I know it is a subset of HTML.